src/connection.cpp

```diff
@@ -191,15 +217,106 @@ namespace oxen::quic
         return "{:02x}"_format(fmt::join(std::begin(data), std::begin(data) + datalen, ""));
     }
 
-    uint64_t Connection::next_dgram_id()
+    void Connection::packet_io_ready()
     {
-        _last_dgram_id += _increment;
-        return _last_dgram_id;
+        event_active(packet_io_trigger.get(), 0, 0);
     }
 
-    void Connection::packet_io_ready()
+    void Connection::handle_conn_packet(const Packet& pkt)
     {
-        event_active(packet_io_trigger.get(), 0, 0);
+        if (auto rv = ngtcp2_conn_in_closing_period(*this); rv != 0)
+        {
+            log::trace(log_cat, "Note: CID-{} in closing period; signaling endpoint to delete connection", scid());
+
+            _endpoint.call([&]() {
+                log::debug(
+                        log_cat, "Error: connection (CID: {}) is in closing period; endpoint deleting connection", scid());
+                _endpoint.delete_connection(scid());
+            });
+            return;
+        }
+
+        if (is_draining())
+        {
+            log::debug(log_cat, "Error: connection is already draining; dropping");
+        }
+
+        // TODO: if read packet gives us failure, should we close?
```

Comment:
This appears to be covered within `read_packet`

Reply:
good catch thats an old comment